src/main/java/com/amazon/opendistroforelasticsearch/sql/executor/format/SelectResultSet.java

```diff
@@ -366,9 +379,55 @@ private String getFieldName(Field field) {
             }
         }
 
+        if (isSelectAllOnly(query)) {
+            populateAllNestedFields(columns, fields);
+        }
+
         return columns;
     }
 
+    private boolean isFieldSpecifiedExplicitly(Map<String, Field> fieldMap, String fieldName) {
+        return fieldMap.containsKey(fieldName);
+    }
+
+    /**
+     *  Verify if field is property field (object or nested ) and matched by wildcard pattern given in SELECT
+     *  A special case is text field with nested keyword. Ignore it in the case of SELECT * ?
+     */
+    private boolean isFieldPropertyAndMatchWildcard(Map<String, Field> fieldMap, String fieldName) {
+        int lastDot = fieldName.lastIndexOf(".");
+        if (lastDot > -1) {
+            String path = fieldName.substring(0, lastDot);
+            return !fieldName.endsWith(".keyword") && fieldMap.containsKey(path + ".*");
```

Comment:
just curious, does es prohibit having a field with name **keyword** ?

Reply:
> just curious, does es prohibit having a field with name **keyword** ?

I'm not sure about a regular field. But in this case, `endsWith(".keyword")` is not always the case. ES will do this automatically if you don't specify it beforehand. But customer can do this themselves with any other name. 

For example, "keyword" (not the one after "type") could be anything.

```
            "author": {
              "type": "text",
              "fields": {
                "keyword": {
                  "type": "keyword",
                  "ignore_above" : 256
                }
              }
            }
```

That is why the proposal in the Description of PR to build domain layer on top of another ES mapping query API.